File: backend/java/localpos-user-service/src/main/java/com/pds/localpos/userservice/service/impl/RoleServiceImpl.java
package com.pds.localpos.userservice.service.impl;

import com.pds.localpos.userservice.dto.response.RoleResponse;
import com.pds.localpos.userservice.mapper.RoleMapper;
import com.pds.localpos.userservice.repository.RoleRepository;
import com.pds.localpos.userservice.service.RoleService;
import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.stereotype.Service;

import java.util.Set;
import java.util.stream.Collectors;

@Service
@RequiredArgsConstructor
@Slf4j
public class RoleServiceImpl implements RoleService {

    private final RoleRepository roleRepository;

    @Override
    public Set<RoleResponse> getAllRoles() {
        log.info("Fetching all roles");
        Set<RoleResponse> roles = roleRepository.findAll().stream()
                .map(RoleMapper::toDTO)
                .collect(Collectors.toSet());
        log.info("Fetched {} roles", roles.size());
        return roles;
    }
}

File: backend/kotlin/localpos-user-service-kt/src/main/kotlin/com/pds/localpos/userservice/service/impl/RoleServiceImpl.kt
package com.pds.localpos.userservice.service.impl

import com.pds.localpos.userservice.dto.RoleDTO
import com.pds.localpos.userservice.mapper.toDTO
import com.pds.localpos.userservice.repository.RoleRepository
import com.pds.localpos.userservice.service.RoleService
import org.springframework.stereotype.Service

@Service
class RoleServiceImpl(private val roleRepository: RoleRepository) : RoleService {

    override fun getAllRoles(): Set<RoleDTO> {
        return roleRepository.findAll()
            .map { it.toDTO() }
            .toSet()
    }
}